### PR TITLE
Host-config with OpenMP and CUDA enabled

### PR DESCRIPTION
**Pull request by bmhan12**
This PR:

- Adds a gcc@8.3.1 host-config with both openmp and cuda enabled.
- Relates to #520 

EDIT: 

- Adds openmp to the clang@8+xlf+cuda host-config

### CODE REVIEW

File: .gitlab/build_lassen.yml [hunk 1]
```diff
@@ -57,6 +57,12 @@ lassen-gcc_7_3_1-src:
     HOST_CONFIG: "lassen-blueos_3_ppc64le_ib_p9-${COMPILER}.cmake"
   extends: [.src_build_on_lassen]
 
+lassen-gcc_8_3_1_cuda-src:
```
Comment (rhornung67) [inline .gitlab/build_lassen.yml:60]: I think it would be good to also include the cuda version in the job labels. Do others agree?

**Review by kennyweiss — APPROVED** [review]
Thanks @bmhan12 for getting this to work.

When I filed #520, I was hoping for a solution that has OpenMP and CUDA for our clang+cuda and/or XL+cuda builds since these are the main compilers of interest on this platform. Looks like I should have been more specific.

**Comment by bmhan12** [comment]
> I was hoping for a solution that has OpenMP and CUDA for our clang+cuda and/or XL+cuda builds...

Yes, initially I tried adding OpenMP to the clang+cuda config, but was running into the same problem @white238 ran into documented in the issue (have not tried XL + cuda, I'll look into that).

This configuration surprisingly worked, so I figured it was better to have a config that tested openmp+cuda then not.

**Comment by bmhan12** [comment]
For future self-reference:

Punted on adding `+openmp` to`xl+cuda` host-config.
Ran into the following error building the resulting `xl+openmp+cuda` generated config in `debug` mode:

```
axom/src/thirdparty/axom/fmt/chrono.h:17:
axom/src/thirdparty/axom/fmt/format.h:373:16: warning: unknown attribute 'no_sanitize' ignored
      [-Wunknown-attributes]
__attribute__((no_sanitize("undefined")))
               ^
1 warning generated.
    1500-030: (I) INFORMATION: axom::fmt::v7::detail::fallback_format<long double>(long double, int, bool, buffer<char> &, int &): Additional optimization may be attained by recompiling and specifying MAXMEM option with a value greater than 8192.
    1500-004: (U) INTERNAL COMPILER ERROR while compiling axom::fmt::v7::detail::dragonbox::cache_accessor<double>::get_cached_power(int).  Compilation ended.  Contact your Service Representative and provide the following information: Internal abort. For more information visit: http://www.ibm.com/support/docview.wss?uid=swg21110810
1586-346 (U) An error occurred during code generation.  The code generation return code was 1.
```

@cyrush Reported conduit-overlink has ran into the same error message.

Attempted to build the same config in `release` mode.
Result was an almost complete compilation in ~6 hours of compilation time with some failing tests.

**Comment by rhornung67** [comment]
Isn't XL a fun compiler to use? 😄

**Comment by kennyweiss** [comment]
Thanks for digging into this @bmhan12 

Re:
```cpp
 __attribute__((no_sanitize("undefined")))
```
Looks like that was already an issue for clang, and we added an `ifdef` guard for it.

https://github.com/LLNL/axom/blob/c9451e2cfeaf9eb7bcadce2effbc08c5513c314f/src/thirdparty/axom/fmt/format.h#L371-L374

Presumably, we can do the same for `XL` if we choose to. Given the compilation times, I'm not sure if it's worth the effort until users want/need it.

File: src/axom/slam/examples/lulesh2.0.3/CMakeLists.txt [hunk 2]
```diff
@@ -43,6 +43,13 @@ blt_add_executable(
 
 if(AXOM_ENABLE_TESTS)
     if(ENABLE_MPI)
+
+        # CMake inserts Fortran flag that forces one OpenMP thread when
+        # compiling with MPI and xlf
+        if(ENABLE_FORTRAN)
+          list(REMOVE_ITEM CMAKE_Fortran_IMPLICIT_LINK_LIBRARIES "xlomp_ser")
+        endif()
+
```
[the remark this answers is not in the document]
Comment (white238) [reply src/axom/slam/examples/lulesh2.0.3/CMakeLists.txt:52]: I can see the use in that...

**Pull request merged**